src/io/output_element.hh

```diff
@@ -0,0 +1,290 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * 
+ * @file    output_element.hh
+ * @brief   Class OutputElement and its iterator OutputElementIterator on the output mesh.
+ */
+
+#ifndef OUTPUT_ELEMENT_HH_
+#define OUTPUT_ELEMENT_HH_
+
+#include "output_mesh.hh"
+
+template <int spacedim>
+class ElementAccessor;
+
+
+#include "mesh/accessors.hh"
+#include "mesh/mesh.h"
+#include "mesh/point.hh"
+#include "output_mesh.hh"
+
+
+/** @brief Represents an element of the output mesh.
+ * Provides element access on the data of the output mesh (nodes, connectivity, offsets etc.).
+ * 
+ * Vertex function suppose spacedim = 3 at the moment.
+ */
+class OutputElement
```

Comment:
I solved the first issue.
The second one:
To be honest, I don't understand much the first two options you named :(
And I don't like the third one either.. we would 3 version of the mesh saved side by side at some times...